pex/pip.py

```diff
@@ -729,33 +751,93 @@ def _fixup_install(
         )
-        if not direct_url_relpath:
+        if not record_relpath:
             return
 
-        direct_url_abspath = os.path.join(install_dir, direct_url_relpath)
-        if not os.path.exists(direct_url_abspath):
-            return
+        exclude_relpaths = []
+
+        record_abspath = os.path.join(install_dir, record_relpath)
+        dist_info_dir = os.path.dirname(record_abspath)
 
-        with open(direct_url_abspath) as fp:
-            if urlparse.urlparse(json.load(fp)["url"]).scheme != "file":
-                return
+        direct_url_abspath = os.path.join(dist_info_dir, "direct_url.json")
+        direct_url_relpath = os.path.relpath(direct_url_abspath, install_dir)
+        if os.path.exists(direct_url_abspath):
+            with open(direct_url_abspath) as fp:
+                if urlparse.urlparse(json.load(fp)["url"]).scheme == "file":
+                    exclude_relpaths.append(os.path.relpath(direct_url_abspath, install_dir))
+                    os.unlink(direct_url_abspath)
 
-        os.unlink(direct_url_abspath)
+        to_rehash = {}
+        if modified_scripts:
+            for modified_script in modified_scripts:
+                # N.B.: Pip installs wheels with RECORD entries like `../../bin/script` even when it's
+                # called in `--target <dir>` mode which installs the script in `bin/script`.
+                record_relpath = os.path.join(os.pardir, os.pardir, modified_script)
+                modified_script_abspath = os.path.join(install_dir, modified_script)
+                to_rehash[record_relpath] = modified_script_abspath
 
         # The RECORD is a csv file with the path to each installed file in the 1st column.
         # See: https://www.python.org/dev/peps/pep-0376/#record
-        dist_info_dir = os.path.dirname(direct_url_abspath)
-        with closing(
-            fileinput.input(files=[os.path.join(dist_info_dir, "RECORD")], inplace=True)
-        ) as record_fi:
-            for line in record_fi:
-                if line.split(",")[0] != direct_url_relpath:
+        with closing(fileinput.input(files=[record_abspath], inplace=True, mode="rU")) as record_fi:
+            csv_writer = None  # type: Optional[CSVWriter]
+            for path, existing_hash, existing_size in csv.reader(
+                record_fi, delimiter=",", quotechar='"'
+            ):
+                if csv_writer is None:
+                    # N.B.: The raw input lines include a newline that varies between '\r\n' and
+                    # '\n' when the wheel was built from an sdist by Pip depending on whether the
+                    # interpreter used was Python 2 or Python 3 respectively. As such, we normalize
+                    # all RECORD files to use '\n' regardless of interpreter.
+                    csv_writer = cast(
+                        "CSVWriter",
+                        csv.writer(sys.stdout, delimiter=",", quotechar='"', lineterminator="\n"),
```

Comment:
The `sys.stdout` here should have raised the same alarm bells as the `print` for example.